src/ParametricOptInterface.jl

```diff
@@ -364,7 +364,19 @@ function MOI.set(model::Optimizer, ::MOI.Name, name::String)
     return MOI.set(model.optimizer, MOI.Name(), name)
 end
 function MOI.get(model::Optimizer, ::MOI.ListOfVariableIndices)
-    return MOI.get(model.optimizer, MOI.ListOfVariableIndices())
+    return _all_variables(model)
+end
+function MOI.get(
+    model::MOI.Utilities.CachingOptimizer,
+    ::MOI.ListOfVariableIndices,
+)
```

Comment:
🏴‍☠️ Please don't pirate a method like this 🏴‍☠️

I think it needs a new attribute, because `ListOfVariableIndices` needs to return all variables added to the model.

Reply:
Agreed.
ListOfPureVariables and ListOfParameters is the way.